lineman/app/components/group_page/group_page.haml

```diff
@@ -18,10 +18,7 @@
               %group_actions_dropdown{group: 'groupPage.group'}
             .lmo-btn-wrapper
               %notification_volume_dropdown{translate-root: 'group_page', group: 'groupPage.group'}
-          .group-page__nonmember-actions{ng-if: '!groupPage.isMember()'}
-            %a.btn.btn-success.btn-block.lmo-btn-icon{href: '', ng-click: 'groupPage.joinGroup()'}
-              %i.fa.fa-lg.fa-plus{aria-hidden: 'true'}>
-              %span{translate: 'group_page.join_group' }
+          %join_group_button{group: 'groupPage.group'}
```

Comment:
I think I strongly prefer that the directive itself decides if it should display. It means you don't need "should this directive run" logic on the parent directive. Thus more self contained. It also means that as a user of the directive you don't need to know anything except what arguments it takes.

I think the reason ng-if ends up outside the directive is because it seems to break a directive if it is on the first element... thus the `.blank`.

I'm open to these being two buttons though.